aip/0123.md

```diff
@@ -72,6 +72,60 @@ message Topic {
 - Patterns **must** correspond to the [resource name][aip-122].
 - Pattern variables (the segments within braces) **must** use `snake_case`, and
   **must not** use an `_id` suffix.
+- Pattern variables (the segments within braces) **must** begin with a
+lower-cased letter and contain only ASCII letters, numbers, and underscore
+(`/[a-z][a-zA-Z0-9_]*/`).
+
+#### Complex resource ID path segments
```

Comment:
@jskeet What are your thoughts here? I think I would prefer not to bless this, and to _only_ have the client libraries AIP cover it (and this would not be accepted usage overall).

At minimum, this feels like a "should not".

Reply:
I'm certainly happy to discourage it, but probably with a note to explain that it's supported for historical reasons in existing APIs. There should be a high bar for adding any new APIs that use this functionality.

Reply:
Right, that seems correct to me. We also need to point to what people _should_ do, which we conveniently already describe in [AIP-124](https://aip.dev/124#many-to-many-associations).

Reply:
Updated with a note to discourage use, and point to AIP-124.

Reply:
I am still not super comfortable with this. Can we put this in the client libraries AIP and not in this one?

Reply:
I tend to agree. Moved this section into AIP-4231.